src/test/func/external_pagemap/external_pagemap.cc

```diff
@@ -0,0 +1,21 @@
+#define SNMALLOC_EXPOSE_PAGEMAP 1
+#  include <override/malloc.cc>
+
+int main()
+{
+	auto &p = ExternalGlobalPagemap::pagemap();
+	auto &global = GlobalPagemap::pagemap();
+	assert(&p == &global);
```

Comment:
Can you use `SMALLOC_CHECK` in the test please.  This should be checked in release builds.

Reply:
`SNMALLOC_CHECK` was added after this PR was originally raised.  I've rebased it on master.